README.md

```diff
@@ -10,37 +10,159 @@ This is a port of [git-machete](https://github.com/VirtusLab/git-machete#git-mac
-gitmachete.frontend.ui
-```
+### Where to find this plugin window
 
-The standard practice in Java logging is to use fully-qualified name of each class as category;
-in our case, however, we're only ever using the categories provided above.
-FQCN (and method name), however, is always a part of the log message itself.
+Git Machete IntelliJ Plugin is available under `Git` section (in the bottom of the IDE) in the `Git Machete` tab.
+You can also use `Ctrl` + `Alt` + `Shift` + `M` shortcut to open this plugin.
 
-By default, IntelliJ logs everything with level `INFO` and above into `idea.log` file. <br/>
-The exact location depends on a specific IntelliJ installation; check `Help` -> `Show Log in Files` to find out. <br/>
-Tip: use `tail -f` to watch the log file as it grows.
+![](docs/open_git_machete.gif)
 
-To enable logging in `DEBUG` level, add selected categories to list in `Help` -> `Diagnostic Tools` -> `Debug Log Settings`. <br/>
-A relatively small amount of `TRACE`-level logs is generated as well.
+### Machete file editor
 
+If you are familiar with console equivalent of this tool ([git-machete](https://github.com/VirtusLab/git-machete#git-machete)) then for sure you know what is the `machete` file.
+For those who don't know this yet, `machete` file is located inside `.git` directory in your repo.
+It describes relation between branches in your repository (these relations are probably determined by order of branch creation - which branch form which - but this is not a rule).
+For more information look at the [reference blog post](https://medium.com/virtuslab/make-your-way-through-the-git-rebase-jungle-with-git-machete-e2ed4dbacd02). <br/>
+Machete file editor will help you to manage `machete` file: it underlines any errors (bad indentation or nonexistent branches) and propose branch names based on local repository branches.
+When file editing is done, you can click button in top right corner of file editor to refresh machete branch tree.
 
-## Development
+![](docs/machete_file_editor.gif)
+
+### Checking out branches
+
+With Git Machete you can easily check out branches that have been defined in `machete` file.
+Just right click on the branch you wont to check out and select `Checkout Branch`.
+You can also double-click on the selected branch to check it out.
+Current branch is underlined in branch tree.
+
+![](docs/checkout.gif)
+
+### Listing commits
+
+Git Machete can display commits belonging to the branches.
+It displays only these commits that are unique for the selected branch.
+When you click on eye button on the left side of plugin, you can toggle between show and hide commits.
+
+![](docs/toggle_listing_commits.gif)
+
+
+### Rebase
+
+When branch tree is created, Git Machete indicates relation between parent branch and each of its child branches.
+If edge between them is green that mean child branch is in-sync with its parent branch - in other words there are no commits in parent branch that doesn't belong to the child.
```

Comment:
```suggestion
If edge between them is green that means child branch is in sync with its parent branch - in other words there are no commits in parent branch that don't belong to the child.
```
Btw that's a pretty good & intuitive wording - I always had a problem how to convey `in sync` more clearly